src/databricks/labs/ucx/assessment/crawlers.py

```diff
@@ -250,3 +269,150 @@ def snapshot(self) -> list[ClusterInfo]:
         for row in self._fetch(f"SELECT * FROM {self._schema}.{self._table}"):
             yield JobInfo(*row)
+
+
+class ExternallyOrchestratedJobTaskCrawler(CrawlerBase):
+    def __init__(self, ws: WorkspaceClient, sbe: SqlBackend, schema):
+        super().__init__(sbe, "hive_metastore", schema, "job_runs", ExternallyOrchestratedJobTask)
+        self._ws = ws
+
+    def _crawl(self) -> list[ExternallyOrchestratedJobTask]:
+        no_of_days_back = datetime.timedelta(days=90)  # todo make configurable in yaml?
+        start_time_from = datetime.datetime.now() - no_of_days_back
+        # todo figure out if we need to specify a default timezone
+        all_job_runs = list(
+            self._ws.jobs.list_runs(
+                expand_tasks=True,
+                start_time_from=start_time_from,
+                start_time_to=datetime.datetime.now(),
+                run_type=RunType.SUBMIT_RUN,
+            )
+        )
+        all_jobs = list(self._ws.jobs.list())
+        all_clusters: dict[str, ClusterDetails] = {c.cluster_id: c for c in self._ws.clusters.list()}
+        return self._assess_job_runs(all_clusters, all_job_runs, all_jobs)
+
+    def _create_hash_from_job_run_task(self, task: RunTask) -> str:
+        """
+        Check all hashable attributes, append to a singular list, remove None's,
+        pipe separate and hash the value to provide a uniqueness identifier
+        - specifically ignore parameters as these change.
+        """
+        hash_values = []
+        if task.notebook_task is not None:
+            hash_values.append(task.notebook_task.notebook_path)
+        if task.spark_python_task is not None:
+            hash_values.append(task.spark_python_task.python_file)
+        if task.spark_submit_task is not None:
+            hash_values.append(task.spark_submit_task.parameters)
+        if task.spark_jar_task is not None:
+            hash_values.append(task.spark_jar_task.jar_uri)
+            hash_values.append(task.spark_jar_task.main_class_name)
+        if task.pipeline_task is not None:
+            hash_values.append(task.pipeline_task.pipeline_id)
+        if task.python_wheel_task is not None:
+            hash_values.append(task.python_wheel_task.package_name)
+            hash_values.append(task.python_wheel_task.entry_point)
+        if task.sql_task is not None:
+            hash_values.append(task.sql_task.file.path)
+            hash_values.append(task.sql_task.alert.alert_id)
+            hash_values.append(task.sql_task.dashboard.dashboard_id)
+            hash_values.append(task.sql_task.query.query_id)
+        if task.dbt_task is not None:
+            task.dbt_task.commands.sort()
+            hash_values.append(task.dbt_task.schema)
+            hash_values.append(task.dbt_task.catalog)
+            hash_values.append(task.dbt_task.warehouse_id)
+            hash_values.append(task.dbt_task.project_directory)
+            hash_values.append(",".join(task.dbt_task.commands))
```

Comment:
```suggestion
            hash_values.append(",".join(sorted(task.dbt_task.commands)))
```
this might be a bit more deterministic

Reply:
okay, will opt for this instead